libs/core/tests/unit/mutex.cpp

```diff
@@ -27,51 +27,54 @@
 
 #ifdef FETCH_DEBUG_MUTEX
 
+using fetch::Mutex;
+using fetch::RMutex;
+
 TEST(DebugMutex, SimpleProblem)
 {
-  fetch::MutexRegister::ThrowOnDeadlock();
+  fetch::DeadlockHandler::ThrowOnDeadlock();
   {
-    fetch::DebugMutex                  mutex;
-    std::lock_guard<fetch::DebugMutex> guard1(mutex);
-    EXPECT_THROW(std::lock_guard<fetch::DebugMutex> guard2(mutex), std::runtime_error);
+    Mutex                  mutex;
+    std::lock_guard<Mutex> guard1(mutex);
+    EXPECT_THROW(std::lock_guard<Mutex> guard2(mutex), std::runtime_error);
   }
 
   {
-    fetch::DebugMutex                  mutex1;
-    fetch::DebugMutex                  mutex2;
-    std::lock_guard<fetch::DebugMutex> guard1(mutex1);
-    std::lock_guard<fetch::DebugMutex> guard2(mutex2);
+    Mutex                  mutex1;
+    Mutex                  mutex2;
+    std::lock_guard<Mutex> guard1(mutex1);
+    std::lock_guard<Mutex> guard2(mutex2);
 
-    EXPECT_THROW(std::lock_guard<fetch::DebugMutex> guard3(mutex2), std::runtime_error);
+    EXPECT_THROW(std::lock_guard<Mutex> guard3(mutex2), std::runtime_error);
   }
 }
 
 TEST(DebugMutex, MultiThreadDeadlock)
 {
-  fetch::MutexRegister::ThrowOnDeadlock();
-  fetch::DebugMutex m[5];
-  auto              f = [&m](int32_t n) {
-    std::lock_guard<fetch::DebugMutex> guard1(m[n]);
+  fetch::DeadlockHandler::ThrowOnDeadlock();
```

Comment:
Seems odd to do it this way - the user will not know if both throw on deadlock and abort on deadlock are set what the priority would be. Wouldn't it make more sense to do `DeadlockHandler::BehaviourOnDeadlock(DeadlockBehaviour::Throw)`?

Reply:
Well, it would be a refactor of current behaviour on itself, unsure if should go in the same PR.